syntax = "proto3";

package meter;

message ReadingPacket {
  repeated ReadingMessage Readings = 1;
  ReadingStatus Successful = 2;
}

service MeterReadingService {
  rpc AddReading (ReadingPacket) returns (StatusMessage);
}

message ReadingMessage {
  int32 CustomerId = 1; //ordinal position in the serialization
  int32 ReadingValue = 2;
  string Notes = 3;
  string ReadingTime = 5;
  reserved 4;
  reserved 'Successful';
}

message StatusMessage {
  ReadingStatus Success = 1;
  string Message = 2;
}

enum ReadingStatus {
  Unknown = 0; //this are actual values
  Success = 1;
  Failure = 2;
}